google/cloud/pubsub/subscriber_connection.cc

```diff
@@ -85,78 +99,39 @@ class SubscriberConnectionImpl : public pubsub::SubscriberConnection {
       std::lock_guard<std::mutex> lk(mu_);
       auto constexpr kLength = 32;
       auto constexpr kChars = "abcdefghijklmnopqrstuvwxyz0123456789";
-      return google::cloud::internal::Sample(generator_, kLength, kChars);
+      return internal::Sample(generator_, kLength, kChars);
     }();
-    return CreateSubscriptionSession(
-        subscription_, options_, stub_, background_->cq(), std::move(client_id),
-        std::move(p), retry_policy_->clone(), backoff_policy_->clone());
+    return CreateSubscriptionSession(subscription_, opts_, stub_,
+                                     background_->cq(), std::move(client_id),
+                                     std::move(p));
   }
 
  private:
   pubsub::Subscription const subscription_;
-  pubsub::SubscriberOptions const options_;
+  Options const opts_;
   std::shared_ptr<pubsub_internal::SubscriberStub> stub_;
   std::shared_ptr<BackgroundThreads> background_;
-  std::unique_ptr<pubsub::RetryPolicy const> retry_policy_;
-  std::unique_ptr<pubsub::BackoffPolicy const> backoff_policy_;
   std::mutex mu_;
-  google::cloud::internal::DefaultPRNG generator_;
+  internal::DefaultPRNG generator_;
 };
 }  // namespace
 
 std::shared_ptr<pubsub::SubscriberConnection> MakeSubscriberConnection(
-    pubsub::Subscription subscription, pubsub::SubscriberOptions options,
-    pubsub::ConnectionOptions connection_options,
-    std::shared_ptr<SubscriberStub> stub,
-    std::unique_ptr<pubsub::RetryPolicy const> retry_policy,
-    std::unique_ptr<pubsub::BackoffPolicy const> backoff_policy) {
-  std::vector<std::shared_ptr<SubscriberStub>> children{std::move(stub)};
-  return MakeSubscriberConnection(std::move(subscription), std::move(options),
-                                  std::move(connection_options),
-                                  std::move(children), std::move(retry_policy),
-                                  std::move(backoff_policy));
-}
-
-std::shared_ptr<pubsub::SubscriberConnection> MakeSubscriberConnection(
-    pubsub::Subscription subscription, pubsub::SubscriberOptions options,
-    pubsub::ConnectionOptions connection_options,
-    std::vector<std::shared_ptr<SubscriberStub>> stubs,
-    std::unique_ptr<pubsub::RetryPolicy const> retry_policy,
-    std::unique_ptr<pubsub::BackoffPolicy const> backoff_policy) {
-  auto default_retry_policy = [] {
-    // Subscribers are special: by default we want to retry essentially forever
```

Comment:
Where did this code go?

Reply:
Sorry, this was confusing because of how I broke up the PRs. But the replacement code was introduced in #7335.

We default the subscriber only retry policy value here:
https://github.com/googleapis/google-cloud-cpp/blob/6112ac217b43961b2f19accac6070fae1f05f93c/google/cloud/pubsub/internal/defaults.cc#L132-L141

and test for the value here:
https://github.com/googleapis/google-cloud-cpp/blob/6112ac217b43961b2f19accac6070fae1f05f93c/google/cloud/pubsub/internal/defaults_test.cc#L168-L171

Reply:
And our entry point calls `DefaultSubscriberOptions()` https://github.com/googleapis/google-cloud-cpp/blob/9479521e1fd47167a5849f5c42cad114d7410ce1/google/cloud/pubsub/subscriber_connection.cc#L54

Reply:
Ack, SGTM